src/components/LGridLayer.vue

```diff
@@ -0,0 +1,89 @@
+<script>
+import { onMounted, onUnmounted, ref, inject, nextTick, h, render } from "vue";
+import { remapEvents, propsBinder } from "../utils.js";
+import {
+  props as gridLayerProps,
+  setup as gridLayerSetup,
+} from "../functions/gridLayer";
+
+export default {
+  props: {
+    ...gridLayerProps,
+    childRender: {
+      type: Function,
+      required: true,
+    },
+  },
+  setup(props, context) {
+    const leafletRef = ref({});
+    const tileComponents = ref({});
+    const root = ref(null);
+    const ready = ref(false);
+
+    const addLayer = inject("addLayer");
+
+    const { options, methods } = gridLayerSetup(props, leafletRef);
+
+    onMounted(async () => {
+      const { GridLayer, DomEvent } = await import(
+        "leaflet/dist/leaflet-src.esm"
+      );
+
+      methods.onUnload = (e) => {
+        const key = leafletRef.value._tileCoordsToKey(e.coords);
+        if (tileComponents[key]) {
+          tileComponents[key].innerHTML = "";
+          tileComponents[key] = undefined;
+        }
+      };
+
+      methods.setTileComponent = () => {
+        leafletRef.value.redraw();
+      };
+
+      const GLayer = GridLayer.extend({
+        createTile(coords) {
+          const key = leafletRef.value._tileCoordsToKey(coords);
+          tileComponents[key] = document.createElement("div");
```

Comment:
I wonder if there will be call to allow users to specify whether to make something other than a div, in particular canvas, but I'm OK with leaving that for a future PR, unless you think it's worth adding at this point.

Reply:
Also, I haven't really looked into what all it handles, but would it be worth using `L.DomUtil.create` here for greater consistency within the map, or not?

Reply:
🤔  Since we allow to pass a render function inside, then the users can add anything to it, I guess a div is just a container for us to function as a `hook` for leaflet. 

I think it allows us to have a little bit more of control to box each layer in the same fixed element, anyhow the users can then put inside anything.

Nice idea with `DomUtil` updateding